libmcount/plthook.c

```diff
@@ -744,6 +749,8 @@ unsigned long plthook_entry(unsigned long *ret_addr, unsigned long child_idx,
 
 	recursion = false;
 
+	// for i386 architecture.
+	child_idx = mcount_arch_child_idx(child_idx);
```

Comment:
Please separate this change as a new commit.

Reply:
I think this change - making `mcount_arch_child_idx()` - should be added before i386 support was enabled.  The actual implementation for i386 will be moved to the initial i386 patch.  Otherwise the initial i386 support is broken at the moment (until this commit is applied) and it'd hurt bisectability.